### DISCUSSION TITLE
How to increase video quality

### DISCUSSION THREAD

**Discussion by changmiao007** [NetrisTV/ws-scrcpy#139] [Q&A]
How to increase the quality of video? It's best to achieve the screen capture quality of Android phones？

**Comment by drauggres** [comment]
Hi.

The client sends a video settings to the server to start the stream. There is `Configure stream` button in the UI that shows dialog, where you can change video settings before start. After start you can change video settings any time from the `More` menu (three dots). In both cases client will send controll message with type `TYPE_CHANGE_STREAM_PARAMETERS`, like here:
https://github.com/NetrisTV/ws-scrcpy/blob/4247f2ae7a47a97d262f7456ad552053692914e4/src/app/googDevice/client/StreamClientScrcpy.ts#L362-L364

Each [player](https://github.com/NetrisTV/ws-scrcpy/discussions/30) has own `preferredVideoSettings` that are used by default.
When you manually change video settings, they are stored into [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) and will be used instead of the default "preferred".
https://github.com/NetrisTV/ws-scrcpy/blob/4247f2ae7a47a97d262f7456ad552053692914e4/src/app/player/BasePlayer.ts#L301

To answer the question, you have these options:
- use `Configure stream` button
- change setting after start from `More` menu
- hardcode new "preferred"

In theory it should be possible to implement automatic quality switching. Players have some quality statistics, you can see them by enabling corresponding option in the `More` menu.


UPD: Under `Configure stream` dialog you can also change [codec options](https://github.com/Genymobile/scrcpy/commit/080a4ee3654a9b7e96c8ffe37474b5c21c02852a) . This could help, but be aware that Broadway.js and TinyH264 support only `Baseline profile` (see https://github.com/NetrisTV/ws-scrcpy/issues/70).

**Comment by drauggres** [comment]
From https://github.com/NetrisTV/ws-scrcpy/issues/140:
>I've seen all the setup related codes here. The video quality still can not meet our requirements. We have set the Java end to send the video stream according to the maximum resolution of the mobile phone, and set the video frame to a fixed size at the front end, but this can only slightly increase the definition, and can not solve the video clarity from the source 

Well, if you already know how to change settings and did it, I'm not sure how else I can help.
Just curious, what screen resolution does the phone have and what video settings did you apply?
